---
title: "Untitled"
author: "Marcel, Owen"
date: "12/2/2021"
output: html_document
---

# Introduction

This markdown contains the analysis of 240 full-length 16S rRNA amplicon sequences, that were performed for the multiple stressor experiment. 

```{r}
library(Biostrings)
library(ShortRead)

library(reshape2)
library(gridExtra)
library(phyloseq)
library(boot)
library(tidyverse)
library(lubridate)
library(googlesheets)
library(here)
library(scales)
library(grid)
library(readxl)
library(ggplot2)
library(dplyr)
library(ggfortify)
library(ggpubr)
library(aod)
library(Rcpp)
library(dada2)
library(microbiome)
library(vegan)
library(plyr)
library(DESeq2)
library(apeglm)
library("ggpubr")
library(patchwork)
library(broom)
library(rstatix)
library("plyr")
library(permute)
library(lattice)
library(tm)
library(stringr)
library(viridis)           


```



# Load phyloseq data, remove chloroplast sequences, remove samples that did not work

```{r}
ps <- readRDS("phyloseq_multiple_stressor.rds")
taxa_names(ps) <- paste0("Seq", seq(ntaxa(ps)))

ps_new <-  subset_taxa(ps, Order != "Chloroplast")

ps_new_1 <- subset_samples(ps_new,Lable != "ori_1")
ps_new_2 <- subset_samples(ps_new_1,Lable != "ori_4")
ps_new_3 <- subset_samples(ps_new_2,Lable != "B_1_24")
ps_new_4 <- subset_samples(ps_new_3,Lable != "C_1_20")
ps_new_5 <- subset_samples(ps_new_4,Lable != "BCD_5_24")
ps_new_6 <- subset_samples(ps_new_5,Lable != "CD_1_24")
ps_new_7 <- subset_samples(ps_new_6,Lable != "BE_5_24")
ps_new_8 <- subset_samples(ps_new_7,Lable != "BCDE_5_24")

ps_new_9 <- subset_samples(ps_new_8,Lable != "CD_5_24")
ps_new_10 <- subset_samples(ps_new_9,Lable != "BE_5_20")
ps_new_11 <- subset_samples(ps_new_10,Lable != "CE_1_24")

ps_out <- subset_samples(ps_new, Lable %in% c("ori_1","ori_4","B_1_24","C_1_20","BCD_5_24","CD_1_24","BE_5_24","BCDE_5_24","CD_5_24"))


```

# Calculation of alpha diversity index
```{r}

diversity_test<- estimate_richness(ps_new_11, split=TRUE, measures=NULL)
diversity_test$File <- rownames(diversity_test)
diversity_test$File <- gsub("\\.\\.", "--", diversity_test$File)
div_raw <- left_join(diversity_test,sample_data(ps_new_11)) 
div_raw <-  div_raw %>%
reorder_levels(Combinations, order = c("Control","F","G","M","A","FG","FM","FA","GM","GA","MA","FGM","FGA","GMA","FMA","FGMA"))%>%
  mutate(Temperature_factor=Temperature_factor)

#filter samples with low amount of reads
div_raw <- div_raw %>%
  filter(Observed >=100) %>%
  filter(Combination != "ori") %>%
  mutate(T2=Temperature^2,T=Temperature_factor) %>%
  select(-Temperature)%>%
  reorder_levels(Combinations, order = c("Control","F","G","M","A","FG","FM","FA","GM","GA","MA","FGM","FGA","GMA","FMA","FGMA"))

stderr <- function(x, na.rm=FALSE) {
  if (na.rm) x <- na.omit(x)
  sqrt(var(x)/length(x))
}

#calculating mean of shannon and observed reads
div_means <- div_raw %>%
  filter(T %in% c("20", "24", "28")) %>%
    dplyr::group_by(Treatment, Combinations, T, Stressor,Stressor_numeric) %>%
    dplyr::summarize(Shannon_mean = mean(Shannon),
                     Observed_mean = mean(Observed),
                     Shannon_SE = stderr(Shannon),
                     Observed_SE = stderr(Observed))%>%
  reorder_levels(Combinations, order = c("Control","F","G","M","A","FG","FM","FA","GM","GA","MA","FGM","FGA","GMA","FMA","FGMA"))


# plots

plot_diversity_all <- div_raw %>%
  filter(T!="19")%>%
ggplot() +
  geom_point(aes(x=Combinations, y=Shannon, shape=Stressor,col=Combinations), size = 3,position = position_dodge(width = 0.4)) +
  facet_wrap("T")+
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold")) +
  ylab ("Shannon index") +
  xlab("Stressor(s) applied") +
   scale_colour_viridis_d(direction = -1)+
   theme(axis.text.x = element_text(angle = 50, hjust = 1))


plot_diversity_means <- div_means %>%
ggplot() +
  geom_point(aes(x=T, y=Shannon_mean, col=Combinations, shape= Stressor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=T,y=Shannon_mean,col=Combinations,ymin=Shannon_mean-Shannon_SE, ymax=Shannon_mean+Shannon_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=T, y=Shannon_mean, col=Combinations),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
 scale_colour_viridis_d(direction = -1)+
  xlab("Temperature [°C]") +
  theme(legend.position = "none") +
  ylab("Diversity_richness (mean)")

plot_diversity_means_numeric <- div_means %>%
ggplot() +
  geom_point(aes(x=Stressor_numeric, y=Shannon_mean, col=T, shape= T), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Stressor_numeric,y=Shannon_mean,col=T,ymin=Shannon_mean-Shannon_SE, ymax=Shannon_mean+Shannon_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold")) +
  xlab("Number of stressors") +
  ylab("Diversity_richness (mean)") +
  theme(legend.position = "none")

#final plot
(plot_diversity_means +  plot_diversity_means_numeric)


#shannon on combinations
anova_diversity <- lm(Shannon ~F*G*M*A*T, data = div_raw)
autoplot(anova_diversity)
hist(residuals(anova_diversity ))
anova(anova_diversity )
summary(anova_diversity)
```

# Calculation rel abundance, just sequences that appear > 0.01 %, making dataframes for each column position

```{r}
ps_rel <- transform_sample_counts(ps_new_11, function(x) x / sum(x) )

relab_threshold <- 0.001

ps_relab <- filter_taxa(ps_rel, function(x) !(sum(x < relab_threshold) == length(x)), TRUE)
ntaxa(ps_new_11)
ntaxa(ps_relab)

ps_relative <- transform_sample_counts(ps_relab, function(x)  x / sum(x))
```

# NMDS 
```{r}
mds_whole <- ps_relative@otu_table %>%
  as.data.frame() %>%
  metaMDS(., 
        distance = "bray",
        k = 3, ## number of dimensions to reduce to
        try = 100, ## number of random starts to try
        autotransform = FALSE ## best not to use
)
 ## 0.11

mds_whole_res <- ps_relative@sam_data %>%
  as.tibble() %>%
  select(Treatment, Stressor,Stressor_numeric, Temperature, Combinations, Temperature_factor, Lable, Name,F,G,M,A) %>%
  bind_cols(as.tibble(scores(mds_whole, display = "sites"))) %>%
  mutate(T2=Temperature^2,T=Temperature_factor) %>%
  select(-Temperature)

mds_whole_res <- mds_whole_res %>%
  filter(Temperature_factor != "19") %>%
  reorder_levels(Combinations, order = c("Control","F","G","M","A","FG","FM","FA","GM","GA","MA","FGM","FGA","GMA","FMA","FGMA"))

ggplot(mds_whole_res, aes(x = NMDS1, y = NMDS2)) +
  geom_point(size = 2) +
  facet_grid(Combinations~Temperature_factor)+
   theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))
```

### Plots of NMDS distances that show high significance
```{r}
#means of NMDS
means_NMDS <- mds_whole_res %>%
  filter(Temperature_factor %in% c("20", "24", "28")) %>%
    dplyr::group_by(Treatment,Combinations, Stressor, Stressor_numeric, Temperature_factor,F,G,M,A) %>%
    dplyr::summarize(NMDS1_mean = mean(NMDS1),
                     NMDS2_mean = mean(NMDS2),
                     NMDS3_mean = mean(NMDS3),
                      NMDS1_SE = stderr(NMDS1),
                     NMDS2_SE = stderr(NMDS2),
                     NMDS3_SE= stderr(NMDS3))
          
```


### NMDS1 and NMDS2 plots

```{r}
#NMDS1 supplement plot

nmds1_plot_temperature <- means_NMDS %>%
ggplot() +
  geom_point(aes(x=Temperature_factor, y=NMDS1_mean, col=Combinations, shape= Stressor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Temperature_factor,y=NMDS1_mean,col=Combinations,ymin=NMDS1_mean-NMDS1_SE, ymax=NMDS1_mean+NMDS1_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=Temperature_factor, y=NMDS1_mean, col=Combinations),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
 scale_colour_viridis_d(direction = -1)+
  xlab("Temperature")  +
   ylab("NMDS1 (mean)") +
  
  theme(legend.position = "none")


plot_NMDS1_means_numeric <- means_NMDS %>%
ggplot() +
  geom_point(aes(x=Stressor_numeric, y=NMDS1_mean, col=Temperature_factor, shape= Temperature_factor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Stressor_numeric,y=NMDS1_mean,col=Temperature_factor,ymin=NMDS1_mean-NMDS1_SE, ymax=NMDS1_mean+NMDS1_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+  
  xlab("Number of stressors") +
  ylab("NMDS1 (mean)") +
  theme(legend.position = "none")


# plot NMDS1
nmds1_plot_temperature +  plot_NMDS1_means_numeric 

#NMDS2 supplement plot

NMDS2_plot_temperature <- means_NMDS %>%
ggplot() +
  geom_point(aes(x=Temperature_factor, y=NMDS2_mean, col=Combinations, shape= Stressor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Temperature_factor,y=NMDS2_mean,col=Combinations,ymin=NMDS2_mean-NMDS2_SE, ymax=NMDS2_mean+NMDS2_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=Temperature_factor, y=NMDS2_mean, col=Combinations),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
 scale_colour_viridis_d(direction = -1)+
  xlab("Temperature") +
  theme(legend.position = "none") +
  ylab("NMDS2 (mean)")


plot_NMDS2_means_numeric <- means_NMDS %>%
ggplot() +
  geom_point(aes(x=Stressor_numeric, y=NMDS2_mean, col=Temperature_factor, shape= Temperature_factor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Stressor_numeric,y=NMDS2_mean,col=Temperature_factor,ymin=NMDS2_mean-NMDS2_SE, ymax=NMDS2_mean+NMDS2_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold")) +
  xlab("Number of stressors") +
  ylab("NMDS2 (mean)")+
  theme(legend.position = "none")

# plot NMDS2
NMDS2_plot_temperature +  plot_NMDS2_means_numeric 

```


###NMDS1 anova and plots
```{r}
anova_NMDS1 <- lm(NMDS1 ~F*G*M*A*T, data = mds_whole_res)
autoplot(anova_NMDS1)
hist(residuals(anova_NMDS1 ))
anova(anova_NMDS1 )
summary(anova_NMDS1)

#F:A:Temperature_factor  

ggplot(means_NMDS) + 
  geom_line(aes(x=Temperature_factor,y=NMDS1_mean, col=as.factor(F),shape=as.factor(A),
group=interaction(as.factor(F),A,M,G)), alpha=0.2)+
  geom_point(mapping = aes(x=Temperature_factor,y=NMDS1_mean, col=as.factor(F),shape=as.factor(A)),size=3)+
  theme_bw() + theme(panel.grid = element_blank())+
  geom_errorbar(aes(x=Temperature_factor,y=NMDS1_mean, ymin=NMDS1_mean-NMDS1_SE,ymax=NMDS1_mean+NMDS1_SE,col=as.factor(F)),width=0.1,alpha=0.2)+
  xlab("Temperature [°C]")



```

###NMDS2 anova and plots
```{r}
#NMDS2 combination 
anova_NMDS2 <- lm(NMDS2 ~F*G*M*A*T, data = mds_whole_res)
autoplot(anova_NMDS2)
hist(residuals(anova_NMDS2 ))
anova(anova_NMDS2 )
summary(anova_NMDS2)
 
```

###NMDS3 anova 
```{r}
#NMDS3

anova_NMDS3 <- lm(NMDS3 ~F*G*M*A*T, data = mds_whole_res)
autoplot(anova_NMDS3)
hist(residuals(anova_NMDS))
anova(anova_NMDS3 )
summary(anova_NMDS3)

anova_NMDS3_numeric <- lm(NMDS3 ~Stressor_numeric*T, data = mds_whole_res)
autoplot(anova_NMDS2_numeric)
hist(residuals(anova_NMDS2_numeric ))
anova(anova_NMDS2_numeric )
summary(anova_NMDS2_numeric)
```

# Visulaization of relative abundance of family level of all combinations with heatmap

```{r}
df_rel <- psmelt(ps_relative)

df_family<- df_rel %>%
  group_by(Family,Lable,Combinations,Temperature_factor) %>%
  dplyr::summarize(Abundance=sum(Abundance))

df_family <- df_family %>%
  reorder_levels(Combinations, order = c("Control","F","G","M","A","FG","FM","FA","GM","GA","MA","FGM","FGA","GMA","FMA","FGMA")) 

df_family%>%
  filter(Family %in% c("Phormidiaceae","Chromatiaceae"),Combinations != "ori") %>%
  ggplot(aes(x=Combinations, y=Abundance, fill=Family)) +
  geom_boxplot()+
  facet_grid("Temperature_factor") +
  theme_bw()

df_family <- df_family %>%
  dplyr::group_by(Family,Combinations,Temperature_factor) %>%
  dplyr::summarize(Abundance_mean = mean(Abundance),
                   Abundance_SE =stderr(Abundance))


index <- which(df_family$Abundance_mean>=0.05)
family_to_keep <- unique(df_family[index,"Family"])
family_to_keep <- unname(unlist(family_to_keep))


df_family $Family_filter <- ifelse(df_family$Family %in% family_to_keep, df_family$Family,"other")

#make other to one
df_family<- df_family %>%
  dplyr::group_by(Family_filter,Combinations,Temperature_factor) %>%
dplyr::summarize(Abundance_sum = sum(Abundance_mean),
                  Abundance_sum_SE =sum(Abundance_SE))


df_family <- df_family %>%
  reorder_levels(Combinations, order = c("Control","F","G","M","A","FG","FM","FA","GM","GA","MA","FGM","FGA","GMA","FMA","FGMA")) 

  cols <-c("Azospirillaceae"="bisque1", "Bacteroidetes_vadinHA17" = "yellow", "Chlorobiaceae"="black","Comamonadaceae" = "green4",  "Hungateiclostridiaceae" ="maroon1", "Lentimicrobiaceae" ="royalblue1","Marinifilaceae"="lightcyan1","Nocardiaceae" ="thistle1", "Phormidiaceae"= "firebrick1", "Prolixibacteraceae" = "magenta4", "Pseudomonadaceae" = "grey","Rhodocyclaceae"="green", "Rhodospillaceae"="grey38","Rubritaleaceae"="brown", "Xanthobacteraceae"="cornflowerblue", "Xanthomonadaceae"="lightskyblue1","Cyanobiaceae"="lavender", "Chromatiaceae" = "lightsalmon")

community_plot <- df_family %>%
  filter(Combinations != "ori",Family_filter !="Mitochondria",Family_filter !="other")%>%
ggplot(aes(x = Combinations, y = Abundance_sum, fill=Family_filter)) + 
    geom_col() +
  facet_wrap("Temperature_factor")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 50, hjust = 1,size=14)) + 
    scale_fill_manual("Family", values = cols) +
  ylab("rel. abundance family level") +
  xlab("Stressor(s) applied")  +
  theme(axis.title=element_text(size=16,face="bold")) +
       theme(legend.text = element_text(size=11)) 

df_family %>%
  filter(Combinations != "ori", Family_filter != "other", Family_filter !="Pirellulaceae", Family_filter !="Cyanobiaceae",Family_filter !="Mitochondria")%>%
ggplot(aes(x = Combinations, y = Family_filter, fill=Abundance_sum)) + 
    geom_tile() +
    scale_fill_gradient2(low="red", mid="white", high="navy", midpoint=0.15, na.value="lightblue",
                 breaks=c(0.05,0.1,0.15,0.2,0.25),
                           limits=c(0.05,0.25)) +
  facet_wrap("Temperature_factor")+
  theme(strip.placement = "inside") +
  theme_bw()+
  theme(axis.text.x = element_text(angle = 50, hjust = 1))


#just phormidiaceae and chromatiaceae for supplementary information

df_family %>%
  filter(Family_filter %in% c("Phormidiaceae","Chromatiaceae"), Combinations!= "ori") %>%
  ggplot() +
  geom_point(aes(x=Combinations,y=Abundance_sum,col=Family_filter,fill=Family_filter),position=position_dodge(width = 0.1),size=4)+
  geom_errorbar(aes(x=Combinations,y=Abundance_sum,col=Family_filter,ymin=Abundance_sum-Abundance_sum_SE, ymax=Abundance_sum+Abundance_sum_SE),width=0.1,alpha=0.4,col="black", position=position_dodge(width = .1))+
  facet_wrap("Temperature_factor")+
  theme_bw() +
  ylab("rel. abundance") +
    theme(axis.text.x = element_text(angle = 50, hjust = 1, size=14)) + 
  labs(col = "Family") + labs(fill="Family")+
  xlab("Stressor combination")
```


# Genus visualization
```{r}
df_genus<- df_rel %>%
  group_by(Genus,Lable,Combinations,Temperature_factor) %>%
  dplyr::summarize(Abundance=sum(Abundance))


df_genus <- df_genus %>%
  dplyr::group_by(Genus,Combinations,Temperature_factor) %>%
dplyr::summarize(Abundance = mean(Abundance))


index <- which(df_genus$Abundance>=0.05)
genus_to_keep <- unique(df_genus[index,"Genus"])
genus_to_keep <- unname(unlist(genus_to_keep))

df_genus $Genus_filter <- ifelse(df_genus$Genus %in% genus_to_keep, df_genus$Genus,"other")

#make other to one
df_genus<- df_genus %>%
  dplyr::group_by(Genus_filter,Combinations,Temperature_factor) %>%
dplyr::summarize(Abundance = sum(Abundance))

df_genus <- df_genus %>%
  reorder_levels(Combinations, order = c("Control","F","G","M","A","FG","FM","FA","GM","GA","MA","FGM","FGA","GMA","FMA","FGMA")) 

genus_plot <- df_genus %>%
  filter(Combinations != "ori", Genus_filter != "other", Genus_filter !="Pirellulaceae", Genus_filter !="Cyanobium_PCC-6307", Genus_filter !="HN-HF0106")%>%
ggplot(aes(x = Combinations, y = Genus_filter, fill=Abundance)) + 
    geom_tile() +
  scale_fill_continuous(name = "rel.abundance",
                      low = "#FF0000",
                      high = "#00CCFF",
                 na.value="white",
                 breaks=c(0.05,0.1,0.15,0.2,0.25),
                           limits=c(0.05,0.25)) +
  facet_wrap("Temperature_factor") +
    theme(axis.text.x = element_text(angle = 50, hjust = 1, size=14)) +
   theme(axis.text.y = element_text(size=12,face="italic"))+ 
  theme(axis.title=element_text(size=16,face="bold")) +
      xlab("Stressor(s) applied") +
      ylab("rel. abundance genus level")
```

```{r}
(community_plot) +
  (plot_spacer() + genus_plot + plot_layout(widths=c(1,100))) +
   plot_layout(ncol=1,heights=c(1,1)) + plot_annotation(tag_levels = "a")
```

# Subpupulation dataset

```{r}
ps_rel <- transform_sample_counts(ps_new_11, function(x) x / sum(x) )

relab_threshold <- 0.001

ps_relab <- filter_taxa(ps_rel, function(x) !(sum(x < relab_threshold) == length(x)), TRUE)
ntaxa(ps_new)
ntaxa(ps_relab)

ps_interest <- subset_taxa(ps_relab, Order %in% c("Cyanobacteriales","Chromatiales","Synechococcales", "Chlorobiales","Leptolyngbyales","Desulfobulbales","Desulfovibrionales","Limnotrichales","Campylobacterales"))

ntaxa(ps_interest)

ps_relative_interest <- transform_sample_counts(ps_interest, function(x)  x / sum(x))

interest <-psmelt(ps_relative_interest)

ps_nmds <- subset_samples(ps_relative_interest, Treatment != "ori")

mds_whole_subset <- ps_nmds@otu_table %>%
  as.data.frame() %>%
  metaMDS(., 
        distance = "bray",
        k = 3, ## number of dimensions to reduce to
        try = 300, ## number of random starts to try
        autotransform = FALSE ## best not to use
) 

## 0.1

mds_whole_res_subset <- ps_nmds@sam_data %>%
  as.tibble() %>%
  select(Treatment,Temperature, Temperature_factor, Stressor,Stressor_numeric, Combinations, Temperature_factor, Lable, Name,F,G,M,A) %>%
  bind_cols(as.tibble(scores(mds_whole_subset, display = "sites"))) %>%
  mutate(T2=Temperature^2,T=Temperature_factor) %>%
  select(-Temperature)

mds_whole_res_subset <- mds_whole_res_subset %>%
  reorder_levels(Combinations, order = c("Control","F","G","M","A","FG","FM","FA","GM","GA","MA","FGM","FGA","GMA","FMA","FGMA"))

# NMDS subset anova combination
#NMDS1 anova combination
anova_NMDS1_mixed <- lm(NMDS1 ~F*G*M*A*T, data = mds_whole_res_subset)
autoplot(anova_NMDS1)
hist(residuals(anova_NMDS1 ))
anova(anova_NMDS1 )
summary(anova_NMDS1)

#NMDS2 anova combination
anova_NMDS2_mixed <- lm(NMDS2 ~F*G*M*A*T, data = mds_whole_res_subset)
autoplot(anova_NMDS2)
hist(residuals(anova_NMDS2 ))
anova(anova_NMDS2 )
summary(anova_NMDS2)

#NMDS3 anova combination
anova_NMDS3 <- lm(NMDS3 ~F*G*M*A*T, data = mds_whole_res_subset)
autoplot(anova_NMDS3)
hist(residuals(anova_NMDS3 ))
anova(anova_NMDS3 )
summary(anova_NMDS3)


#NMDS subset anova numeric

#NMDS3
anova_NMDS3_numeric <- lm(NMDS3 ~ Stressor_numeric*T, data = mds_whole_res_subset)
autoplot(anova_NMDS3_numeric)
hist(residuals(anova_NMDS3_numeric ))
anova(anova_NMDS3_numeric )
summary(anova_NMDS3_numeric)

#means of NMDS

means_NMDS_subset <- mds_whole_res_subset %>%
  filter(Temperature_factor %in% c("20", "24", "28")) %>%
    dplyr::group_by(Treatment,Combinations, Stressor, Stressor_numeric, Temperature_factor,F,G,M,A) %>%
    dplyr::summarize(NMDS1_mean = mean(NMDS1),
                     NMDS2_mean = mean(NMDS2),
                     NMDS3_mean = mean(NMDS3),
                      NMDS1_SE = stderr(NMDS1),
                     NMDS2_SE = stderr(NMDS2),
                     NMDS3_SE= stderr(NMDS3))

#NMDS1 subpopulation

nmds1_plot_temperature_subset <- means_NMDS_subset %>%
ggplot() +
  geom_point(aes(x=Temperature_factor, y=NMDS1_mean, col=Combinations, shape= Stressor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Temperature_factor,y=NMDS1_mean,col=Combinations,ymin=NMDS1_mean-NMDS1_SE, ymax=NMDS1_mean+NMDS1_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=Temperature_factor, y=NMDS1_mean, col=Combinations),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
 scale_colour_viridis_d(direction = -1)+
  xlab("Temperature") +
  ylab("NMDS1_mean subpopulation")+
  theme(legend.position = "none") +
  ylab("NMDS1_subpopulation (mean)")


plot_NMDS1_means_numeric_subset<- means_NMDS_subset %>%
ggplot() +
  geom_point(aes(x=Stressor_numeric, y=NMDS1_mean, col=Temperature_factor, shape= Temperature_factor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Stressor_numeric,y=NMDS1_mean,col=Temperature_factor,ymin=NMDS1_mean-NMDS1_SE, ymax=NMDS1_mean+NMDS1_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
  ylab("NMDS1_subpopulation (mean)") +
  xlab( "Number of stressors")+
  theme(legend.position = "none")

# plot 
nmds1_plot_temperature_subset +  plot_NMDS1_means_numeric_subset 


#NMDS2 subpopulation

NMDS2_plot_temperature_subset <- means_NMDS_subset %>%
ggplot() +
  geom_point(aes(x=Temperature_factor, y=NMDS2_mean, col=Combinations, shape= Stressor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Temperature_factor,y=NMDS2_mean,col=Combinations,ymin=NMDS2_mean-NMDS2_SE, ymax=NMDS2_mean+NMDS2_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=Temperature_factor, y=NMDS2_mean, col=Combinations),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
 scale_colour_viridis_d(direction = -1)+
  xlab("Temperature") +
 theme(legend.position = "none") +
  ylab("NMDS2_subpopulation (mean)")


plot_NMDS2_means_numeric_subset <- means_NMDS_subset %>%
ggplot() +
  geom_point(aes(x=Stressor_numeric, y=NMDS2_mean, col=Temperature_factor, shape= Temperature_factor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Stressor_numeric,y=NMDS2_mean,col=Temperature_factor,ymin=NMDS2_mean-NMDS2_SE, ymax=NMDS2_mean+NMDS2_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
  ylab("NMDS2_subpopulation (mean)") +
  xlab( "Number of stressors")+
 theme(legend.position = "none")

#plot 
NMDS2_plot_temperature_subset +  plot_NMDS2_means_numeric_subset 
```


# Dataset of single genera, namely Tychonema_CCAP_1459-11B, Sulfuricurvum and Chromatium, calculating NMDS for all, respectively
```{r}
ps_relative_cyano <- subset_taxa(ps_relative, Genus ==
                                "Tychonema_CCAP_1459-11B")

df_cyano <- psmelt(ps_relative_cyano)

df_cyano <- df_cyano %>%
  mutate(T2=Temperature^2,T=Temperature_factor) %>%
  select(-Temperature) 

df_cyano <- df_cyano %>%
  filter(Treatment != "ori")

df_cyano <- df_cyano %>%
dplyr::group_by(Treatment,Stressor_numeric,Lable,Combinations, T,F,G,M,A) %>%
dplyr::summarize(Abundance = sum(Abundance)) %>%
  reorder_levels(Combinations, order = c("Control","F","G","M","A","FG","FM","FA","GM","GA","MA","FGM","FGA","GMA","FMA","FGMA"))

#anova on combination
anova_tychonea <- lm(logit(Abundance) ~F*G*M*A*T, data = df_cyano)
autoplot(anova_tychonea)
hist(residuals(anova_tychonea))
anova(anova_tychonea)
summary(anova_tychonea)

# Sulfuricurvum
ps_sulfuri <- subset_taxa(ps_relative, Genus == "Sulfuricurvum")

df_sulfuri <- psmelt(ps_sulfuri)

df_sulfuri <- df_sulfuri %>%
  mutate(T2=Temperature^2,T=Temperature_factor) %>%
  select(-Temperature)

df_sulfuri <- df_sulfuri %>%
  filter(Treatment!="ori") %>%
dplyr::group_by(Treatment,Stressor_numeric,Combinations, Lable,T,F,G,M,A) %>%
dplyr::summarize(Abundance = sum(Abundance))
  
#anova on combinations
anova_sulfuri <- lm(asin(sqrt(Abundance)) ~F*G*M*A*T, data = df_sulfuri)
autoplot(anova_sulfuri)
hist(residuals(anova_sulfuri))
anova(anova_sulfuri)
summary(anova_sulfuri)



```

# Abiotic factors and oxygen
```{r}
abiotic<- ps_relative@sam_data %>%
  as.tibble() %>%
  select(Treatment,Temperature, Combinations, Temperature_factor, Lable, Name,F,G,M,A, Oxygen, TN,TC, pH,Stressor_numeric, Stressor) %>%
  mutate(T2=Temperature^2,T=Temperature_factor) %>%
  select(-Temperature)

pH_anova <- lm(pH ~F*G*M*A*T, data = abiotic)
autoplot(pH_anova)
hist(residuals(pH_anova))
anova(pH_anova)
summary(pH_anova)

oxygen_anova <- lm(Oxygen ~F*G*M*A*T, data = abiotic)
autoplot(oxygen_anova)
hist(residuals(oxygen_anova))
anova(oxygen_anova)
summary(oxygen_anova)

tc_anova <- lm(TC ~F*G*M*A*T, data = abiotic)
autoplot(oxygen_anova)
hist(residuals(oxygen_anova))
anova(oxygen_anova)
summary(oxygen_anova)

tn_anova <- lm(TN ~F*G*M*A*T, data = abiotic)
autoplot(tn_anova)
hist(residuals(tn_anova))
anova(tn_anova)
summary(tn_anova)


#abiotic plots
abiotic <- abiotic %>%
reorder_levels(Combinations, order = c("Control","F","G","M","A","FG","FM","FA","GM","GA","MA","FGM","FGA","GMA","FMA","FGMA"))


stderr <- function(x, na.rm=FALSE) {
  if (na.rm) x <- na.omit(x)
  sqrt(var(x)/length(x))
}

#calculating mean of oxygen and pH 
abiotic_means <- abiotic %>%
  filter(T %in% c("20", "24", "28")) %>%
    dplyr::group_by(Treatment, Combinations, T, Stressor,Stressor_numeric) %>%
    dplyr::summarize(Oxygen_mean = mean(Oxygen),
                     pH_mean = mean(pH),
                     Oxygen_SE = stderr(Oxygen),
                     pH_SE = stderr(pH),
                     TN_mean = mean(TN),
                     TC_mean = mean(TC),
                     TN_SE = stderr(TN),
                     TC_SE = stderr(TC))


plot_oxygen_means <- abiotic_means %>%
ggplot() +
  geom_point(aes(x=T, y=Oxygen_mean, col=Combinations, shape= Stressor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=T,y=Oxygen_mean,col=Combinations,ymin=Oxygen_mean-Oxygen_SE, ymax=Oxygen_mean+Oxygen_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=T, y=Oxygen_mean, col=Combinations),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
 scale_colour_viridis_d(direction = -1)+
  xlab("Temperature") +
  theme(legend.position = "none") +
  ylab("Oxygen[%] (mean)")

plot_oxygen_numeric_means <- abiotic_means %>%
ggplot() +
  geom_point(aes(x=Stressor_numeric, y=Oxygen_mean, col=T, shape= T), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Stressor_numeric,y=Oxygen_mean,col=T,ymin=Oxygen_mean-Oxygen_SE, ymax=Oxygen_mean+Oxygen_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=Stressor_numeric, y=Oxygen_mean, col=T),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
  ylab("Oxygen[%] (mean)") +
  xlab("Number of stressors")+
  theme(legend.position = "none")

#plot oxygen
plot_oxygen_means + plot_oxygen_numeric_means

#pH
plot_pH_means <- abiotic_means %>%
ggplot() +
  geom_point(aes(x=T, y=pH_mean, col=Combinations, shape= Stressor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=T,y=pH_mean,col=Combinations,ymin=pH_mean-pH_SE, ymax=pH_mean+pH_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=T, y=pH_mean, col=Combinations),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
 scale_colour_viridis_d(direction = -1)+
  xlab("Temperature") +
  ylab("pH (mean)")+
 theme(legend.position = "none")

plot_pH_means_numeric <- abiotic_means %>%
ggplot() +
  geom_point(aes(x=Stressor_numeric, y=pH_mean, col=T, shape= T), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Stressor_numeric,y=pH_mean,col=T,ymin=pH_mean-pH_SE, ymax=pH_mean+pH_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=Stressor_numeric, y=pH_mean, col=T),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
  ylab("pH (mean)")+ 
  xlab("Number of stressors")+
  theme(legend.position = "none")
  
#plot pH
plot_pH_means + plot_pH_means_numeric 


#total nitrogen

plot_TN_means <- abiotic_means %>%
ggplot() +
  geom_point(aes(x=T, y=TN_mean, col=Combinations, shape= Stressor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=T,y=TN_mean,col=Combinations,ymin=TN_mean-TN_SE, ymax=TN_mean+TN_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=T, y=TN_mean, col=Combinations),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
 scale_colour_viridis_d(direction = -1)+
  xlab("Temperature") +
  ylab("TN (mean)")+
 theme(legend.position = "none")

plot_TN_means_numeric <- abiotic_means %>%
ggplot() +
  geom_point(aes(x=Stressor_numeric, y=TN_mean, col=T, shape= T), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Stressor_numeric,y=TN_mean,col=T,ymin=TN_mean-TN_SE, ymax=TN_mean+TN_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=Stressor_numeric, y=TN_mean, col=T),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
  ylab("TN (mean)")+ 
  xlab("Number of stressors")+
  theme(legend.position = "none")

#plot nitrogen

plot_TN_means +theme(legend.position = "bottom",
          legend.box = "vertical")  + plot_TN_means_numeric + theme(legend.position = "bottom",
          legend.box = "vertical")

#total carbon 

plot_TC_means <- abiotic_means %>%
ggplot() +
  geom_point(aes(x=T, y=TC_mean, col=Combinations, shape= Stressor), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=T,y=TC_mean,col=Combinations,ymin=TC_mean-TC_SE, ymax=TC_mean+TC_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=T, y=TC_mean, col=Combinations),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
 scale_colour_viridis_d(direction = -1)+
  xlab("Temperature") +
  ylab("TC (mean)")+
 theme(legend.position = "none")

plot_TC_means_numeric <- abiotic_means %>%
ggplot() +
  geom_point(aes(x=Stressor_numeric, y=TC_mean, col=T, shape= T), size = 3, position = position_dodge(width = 0.4)) +
geom_errorbar(aes(x=Stressor_numeric,y=TC_mean,col=T,ymin=TC_mean-TC_SE, ymax=TC_mean+TC_SE), width=0.1,alpha=0.4,position = position_dodge(width = 0.4))+
  geom_line(aes(group=Combinations, x=Stressor_numeric, y=TC_mean, col=T),position = position_dodge(width = 0.4)) +
  theme_bw() + theme(panel.grid = element_blank()) +
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"))+
  ylab("TC (mean)")+ 
  xlab("Number of stressors")+
  theme(legend.position = "none")
  
#plot total carbon

plot_TC_means + plot_TC_means_numeric 
```

# final plot for figure 

```{r}
(plot_diversity_means +  plot_diversity_means_numeric)/
(nmds1_plot_temperature +  plot_NMDS1_means_numeric)/
(nmds1_plot_temperature_subset +  plot_NMDS1_means_numeric_subset)/
(plot_oxygen_means + plot_oxygen_numeric_means)/
(plot_pH_means +theme(legend.position = "bottom",
          legend.box = "vertical",legend.text = element_text(size=12),legend.title = element_text(size=15)) +labs(col = "Stressor combinations",shape= "") + plot_pH_means_numeric + theme(legend.position = "bottom",
          legend.box = "vertical",legend.text = element_text(size=12),legend.title = element_text(size=15))+ labs(col = "Temperature")+ labs(shape= "Temperature")) +plot_annotation(tag_levels = "a") 
```

#supplementary plot of all variables

```{r}
(NMDS2_plot_temperature +  plot_NMDS2_means_numeric) /
(NMDS2_plot_temperature_subset +  plot_NMDS2_means_numeric_subset)/
  (plot_TN_means + plot_TN_means_numeric) /
  (plot_TC_means  +theme(legend.position = "bottom",
          legend.box = "vertical",legend.text = element_text(size=12),legend.title = element_text(size=15)) +labs(col = "Stressor combinations",shape= "") + plot_TC_means_numeric+ theme(legend.position = "bottom",
          legend.box = "vertical",legend.text = element_text(size=12),legend.title = element_text(size=15))+ labs(col = "Temperature") + labs(shape= "Temperature")) + plot_annotation(tag_levels = "a")

```


# extract the f and t test values, make one data frame for all factors

```{r}
#NMDS1
ano_NMDS1_t <- tidy(anova_NMDS1)
names(ano_NMDS1_t)[names(ano_NMDS1_t) == "statistic"] <- "t_statistic"
names(ano_NMDS1_t)[names(ano_NMDS1_t) == "p.value"] <- "t_p.value"
names(ano_NMDS1_t)[names(ano_NMDS1_t) == "term"] <- "treatment_combination"

ano_NMDS1_t <- ano_NMDS1_t %>%
  mutate(term = removeNumbers(treatment_combination))

ano_NMDS1_f <- tidy(anova(anova_NMDS1))
names(ano_NMDS1_f)[names(ano_NMDS1_f) == "statistic"] <- "f_statistic"
names(ano_NMDS1_f)[names(ano_NMDS1_f) == "p.value"] <- "f_p.value"

nmds1 <- full_join(ano_NMDS1_t, ano_NMDS1_f)

nmds1 <- nmds1 %>%
  mutate(color = ifelse(nmds1$f_p.value < 0.05, estimate, NA))

nmds1$variable="NMDS1"

nmds1 <- nmds1 %>%
  filter(term != "(Intercept)", term!="Residuals")

nmds1 <- nmds1 %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))

#NMDS2
ano_NMDS2_t <- tidy(anova_NMDS2)
names(ano_NMDS2_t)[names(ano_NMDS2_t) == "statistic"] <- "t_statistic"
names(ano_NMDS2_t)[names(ano_NMDS2_t) == "p.value"] <- "t_p.value"
names(ano_NMDS2_t)[names(ano_NMDS2_t) == "term"] <- "treatment_combination"

ano_NMDS2_t <- ano_NMDS2_t %>%
  mutate(term = removeNumbers(treatment_combination))

ano_NMDS2_f <- tidy(anova(anova_NMDS2))
names(ano_NMDS2_f)[names(ano_NMDS2_f) == "statistic"] <- "f_statistic"
names(ano_NMDS2_f)[names(ano_NMDS2_f) == "p.value"] <- "f_p.value"

nmds2 <- full_join(ano_NMDS2_t, ano_NMDS2_f)

nmds2 <- nmds2 %>%
  mutate(color = ifelse(nmds2$f_p.value < 0.05, estimate/1.9, NA))

nmds2$variable="NMDS2"

nmds2 <- nmds2 %>%
  filter(term != "(Intercept)", term!="Residuals")

nmds2 <- nmds2 %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))
 
# NMDS1_CB_SB /1.8
ano_nmds1_mixed_t <- tidy(anova_NMDS1_mixed)
names(ano_nmds1_mixed_t )[names(ano_nmds1_mixed_t ) == "statistic"] <- "t_statistic"
names(ano_nmds1_mixed_t )[names(ano_nmds1_mixed_t ) == "p.value"] <- "t_p.value"
names(ano_nmds1_mixed_t )[names(ano_nmds1_mixed_t ) == "term"] <- "treatment_combination"

ano_nmds1_mixed_t   <- ano_nmds1_mixed_t   %>%
  mutate(term = removeNumbers(treatment_combination))

ano_nmds1_mixed_f <- tidy(anova(anova_NMDS1_mixed))
names(ano_nmds1_mixed_f )[names(ano_nmds1_mixed_f ) == "statistic"] <- "f_statistic"
names(ano_nmds1_mixed_f )[names(ano_nmds1_mixed_f ) == "p.value"] <- "f_p.value"

nmds1_mixed <- full_join(ano_nmds1_mixed_t  , ano_nmds1_mixed_f)

nmds1_mixed <- nmds1_mixed %>%
  mutate(color = ifelse(nmds1_mixed$f_p.value < 0.05, estimate/1.8, NA))

nmds1_mixed$variable="NMDS1_CB_SB"

nmds1_mixed <- nmds1_mixed %>%
  filter(term != "(Intercept)", term!="Residuals")

nmds1_mixed <- nmds1_mixed %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))

#NMDS2_CB_SB

ano_nmds2_mixed_t <- tidy(anova_NMDS2_mixed)
names(ano_nmds2_mixed_t )[names(ano_nmds2_mixed_t ) == "statistic"] <- "t_statistic"
names(ano_nmds2_mixed_t )[names(ano_nmds2_mixed_t ) == "p.value"] <- "t_p.value"
names(ano_nmds2_mixed_t )[names(ano_nmds2_mixed_t ) == "term"] <- "treatment_combination"

ano_nmds2_mixed_t   <- ano_nmds2_mixed_t   %>%
  mutate(term = removeNumbers(treatment_combination))

ano_nmds2_mixed_f <- tidy(anova(anova_NMDS2_mixed))
names(ano_nmds2_mixed_f )[names(ano_nmds2_mixed_f ) == "statistic"] <- "f_statistic"
names(ano_nmds2_mixed_f )[names(ano_nmds2_mixed_f ) == "p.value"] <- "f_p.value"

nmds2_mixed <- full_join(ano_nmds2_mixed_t  , ano_nmds2_mixed_f)

nmds2_mixed <- nmds2_mixed %>%
  mutate(color = ifelse(nmds2_mixed$f_p.value < 0.05, estimate, NA))

nmds2_mixed$variable="NMDS2_CB_SB"

nmds2_mixed <- nmds2_mixed %>%
  filter(term != "(Intercept)", term!="Residuals")

nmds2_mixed <- nmds2_mixed %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))

# Richness Shannon
ano_richness_t <- tidy(anova_diversity)
names(ano_richness_t)[names(ano_richness_t) == "statistic"] <- "t_statistic"
names(ano_richness_t)[names(ano_richness_t) == "p.value"] <- "t_p.value"
names(ano_richness_t)[names(ano_richness_t) == "term"] <- "treatment_combination"

ano_richness_t <-ano_richness_t %>%
  mutate(term = removeNumbers(treatment_combination))

ano_richness_f <- tidy(anova(anova_diversity))
names(ano_richness_f )[names(ano_richness_f ) == "statistic"] <- "f_statistic"
names(ano_richness_f)[names(ano_richness_f ) == "p.value"] <- "f_p.value"

richness <- full_join(ano_richness_t, ano_richness_f)

richness <- richness %>%
  mutate(color = ifelse(richness$f_p.value < 0.05, estimate, NA))

richness$variable="Shannon_richness"

richness <- richness %>%
  filter(term != "(Intercept)", term!="Residuals")

richness <- richness %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))

# Tychonema Cyanobacteria /2.2

ano_tychonea_t <- tidy(anova_tychonea)
names(ano_tychonea_t)[names(ano_tychonea_t) == "statistic"] <- "t_statistic"
names(ano_tychonea_t)[names(ano_tychonea_t) == "p.value"] <- "t_p.value"
names(ano_tychonea_t)[names(ano_tychonea_t) == "term"] <- "treatment_combination"

ano_tychonea_t <-ano_tychonea_t %>%
  mutate(term = removeNumbers(treatment_combination))

ano_tychonea_f <- tidy(anova(anova_tychonea))
names(ano_tychonea_f )[names(ano_tychonea_f ) == "statistic"] <- "f_statistic"
names(ano_tychonea_f)[names(ano_tychonea_f ) == "p.value"] <- "f_p.value"

tyochonea <- full_join(ano_tychonea_t, ano_tychonea_f)

tyochonea <- tyochonea %>%
  mutate(color = ifelse(tyochonea$f_p.value < 0.05, estimate/2.2, NA))

tyochonea$variable="Tychonema"

tyochonea <- tyochonea %>%
  filter(term != "(Intercept)", term!="Residuals")

tyochonea <- tyochonea %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))


# Sulfuricurvum /1.27
ano_sulfuri_t <- tidy(anova_sulfuri)
names(ano_sulfuri_t)[names(ano_sulfuri_t) == "statistic"] <- "t_statistic"
names(ano_sulfuri_t)[names(ano_sulfuri_t) == "p.value"] <- "t_p.value"
names(ano_sulfuri_t)[names(ano_sulfuri_t) == "term"] <- "treatment_combination"

ano_sulfuri_t <-ano_sulfuri_t %>%
  mutate(term = removeNumbers(treatment_combination))

ano_sulfuri_f <- tidy(anova(anova_sulfuri))
names(ano_sulfuri_f )[names(ano_sulfuri_f ) == "statistic"] <- "f_statistic"
names(ano_sulfuri_f)[names(ano_sulfuri_f ) == "p.value"] <- "f_p.value"

sulfuri <- full_join(ano_sulfuri_t, ano_sulfuri_f)

sulfuri <- sulfuri %>%
  mutate(color = ifelse(sulfuri$f_p.value < 0.05, estimate/1.27, NA))

sulfuri$variable="Sulfuricurvum"

sulfuri <- sulfuri %>%
  filter(term != "(Intercept)", term!="Residuals")

sulfuri <- sulfuri %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))

#oxygen /9.26

ano_oxygen_t <- tidy(oxygen_anova)
names(ano_oxygen_t )[names(ano_oxygen_t ) == "statistic"] <- "t_statistic"
names(ano_oxygen_t )[names(ano_oxygen_t ) == "p.value"] <- "t_p.value"
names(ano_oxygen_t )[names(ano_oxygen_t ) == "term"] <- "treatment_combination"

ano_oxygen_t  <- ano_oxygen_t  %>%
  mutate(term = removeNumbers(treatment_combination))

ano_oxygen_f <- tidy(anova(oxygen_anova))
names(ano_oxygen_f)[names(ano_oxygen_f) == "statistic"] <- "f_statistic"
names(ano_oxygen_f)[names(ano_oxygen_f) == "p.value"] <- "f_p.value"

oxygen <- full_join(ano_oxygen_t , ano_oxygen_f)

oxygen <- oxygen %>%
  mutate(color = ifelse(oxygen$f_p.value < 0.05, estimate/9.26, NA))

oxygen$variable="oxygen"

oxygen <- oxygen %>%
  filter(term != "(Intercept)", term!="Residuals")

oxygen <- oxygen %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))

#pH /2.62

ano_pH_t <- tidy(pH_anova)
names(ano_pH_t)[names(ano_pH_t) == "statistic"] <- "t_statistic"
names(ano_pH_t)[names(ano_pH_t) == "p.value"] <- "t_p.value"
names(ano_pH_t)[names(ano_pH_t) == "term"] <- "treatment_combination"

ano_pH_t <- ano_pH_t %>%
  mutate(term = removeNumbers(treatment_combination))

ano_pH_f <- tidy(anova(pH_anova))
names(ano_pH_f )[names(ano_pH_f) == "statistic"] <- "f_statistic"
names(ano_pH_f )[names(ano_pH_f) == "p.value"] <- "f_p.value"

pH <- full_join(ano_pH_t , ano_pH_f)

pH <- pH %>%
  mutate(color = ifelse(pH$f_p.value < 0.05, estimate/2.62, NA))

pH$variable="pH"

pH <- pH %>%
  filter(term != "(Intercept)", term!="Residuals")

pH <- pH %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))

#TN/93.77

ano_tn_t <- tidy(tn_anova)
names(ano_tn_t )[names(ano_tn_t ) == "statistic"] <- "t_statistic"
names(ano_tn_t )[names(ano_tn_t ) == "p.value"] <- "t_p.value"
names(ano_tn_t )[names(ano_tn_t ) == "term"] <- "treatment_combination"

ano_tn_t  <- ano_tn_t  %>%
  mutate(term = removeNumbers(treatment_combination))

ano_tn_f <- tidy(anova(tn_anova))
names(ano_tn_f)[names(ano_tn_f) == "statistic"] <- "f_statistic"
names(ano_tn_f)[names(ano_tn_f) == "p.value"] <- "f_p.value"

tn <- full_join(ano_tn_t , ano_tn_f)

tn <- tn %>%
  mutate(color = ifelse(tn$f_p.value < 0.05, estimate/93.77, NA))

tn$variable="tn"

tn <- tn %>%
  filter(term != "(Intercept)", term!="Residuals")

tn <- tn %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))


#TC /65
ano_tc_t <- tidy(tc_anova)
names(ano_tc_t )[names(ano_tc_t ) == "statistic"] <- "t_statistic"
names(ano_tc_t )[names(ano_tc_t ) == "p.value"] <- "t_p.value"
names(ano_tc_t )[names(ano_tc_t ) == "term"] <- "treatment_combination"

ano_tc_t  <- ano_tc_t  %>%
  mutate(term = removeNumbers(treatment_combination))

ano_tc_f <- tidy(anova(tc_anova))
names(ano_tc_f)[names(ano_tc_f) == "statistic"] <- "f_statistic"
names(ano_tc_f)[names(ano_tc_f) == "p.value"] <- "f_p.value"

tc <- full_join(ano_tc_t , ano_tc_f)

tc <- tc %>%
  mutate(color = ifelse(tc$f_p.value < 0.05, estimate/65, NA))

tc$variable="tc"

tc <- tc %>%
  filter(term != "(Intercept)", term!="Residuals")

tc <- tc %>% 
   reorder_levels(treatment_combination, order = c("T24","T28","F", "G","M","A","F:T24","G:T24", "M:T24","A:T24", "F:T28","G:T28","M:T28", "A:T28", "F:G", "F:M", "G:M", "F:A", "G:A", "M:A", "F:G:T24", "F:M:T24", "G:M:T24", "F:A:T24", "G:A:T24", "M:A:T24","F:G:T28", "F:M:T28", "G:M:T28", "F:A:T28", "G:A:T28", "M:A:T28", "F:G:M", "F:G:A","F:M:A","G:M:A","F:G:M:T24","F:G:A:T24","F:M:A:T24", "G:M:A:T24","F:G:M:T28", "F:G:A:T28","F:M:A:T28", "G:M:A:T28","F:G:M:A","F:G:M:A:T24", "F:G:M:A:T28"))

#all abiotic

all_abiotic <- rbind(oxygen, pH, tn, tc)


all_heat <- rbind(tyochonea,sulfuri,nmds1,nmds2,nmds1_mixed, nmds2_mixed,richness,oxygen, pH, tn, tc)

all_heat <- all_heat %>%
  reorder_levels(variable,order=c("Tychonema","Sulfuricurvum","NMDS1","NMDS2","NMDS1_CB_SB","NMDS2_CB_SB","Shannon_richness","oxygen", "pH", "tn", "tc"))
```

## heat map of all factors
```{r}
all_heat %>%
  filter(term != "(Intercept)", term!="Residuals")%>%
ggplot(aes(x = variable, y = treatment_combination, fill=color)) + 
    geom_tile() +
  scale_fill_gradient2(low="navy", mid="white", high="red", 
                       midpoint=0,breaks=c(-1,0,1),
                         na.value="lightgrey",
                           limits=c(-1.2,1.2)
                    ) +
  theme(strip.placement = "inside") +
  theme_bw()+
  theme(axis.text.x = element_text(angle = 50, hjust = 1,size=10)) +
  theme(axis.text.y = element_text(angle = 0, hjust = 1,size=10)) +
  ylab("model term [main effect or interaction]")
```

## just visualization of F:A with all factors

```{r}
all_heat %>%
  filter(treatment_combination %in% c("F","A","F:A","F:A:T24","F:A:T28"), variable != "tn",variable != "tc",color != "NA")%>%
  ggplot()+
    geom_point(aes(x=variable, y=color, shape=treatment_combination,col=treatment_combination),size=4)+
  theme_bw() +
   theme(axis.text=element_text(size=14),
        axis.title=element_text(size=14)) +
  ylab("standardised estimate")+
   theme(axis.text.x = element_text(angle = 50, hjust = 1))+
  geom_hline(yintercept=0)
```





# numeric analysis

slope
```{r}

#oxygen
abiotic<- ps_relative@sam_data %>%
  as.tibble() %>%
  select(Treatment,Temperature, Combinations, Temperature_factor, Lable, Name,F,G,M,A, Oxygen, TN,TC, pH,Stressor_numeric, Stressor)

abiotic <- abiotic %>%
  filter(Treatment!="ori")

abiotic <- abiotic %>%
  mutate(T2=Temperature^2,T=as.factor(Temperature)) %>%
  select(-Temperature)
  

abiotic$Oxygen<- (abiotic$Oxygen-mean(abiotic$Oxygen))/sd(abiotic$Oxygen)
oxygen_anova_numeric <- lm(Oxygen ~T+T:Stressor_numeric, data=abiotic)


oxy_slope<- tidy(oxygen_anova_numeric ,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")

oxy_slope$std.error <- NULL
oxy_slope$statistic <- NULL
oxy_slope$p.value <- NULL


oxy_slope$organisation="ecosystem"
oxy_slope$variable="oxygen"


#pH 

abiotic$pH <- (abiotic$pH-mean(abiotic$pH))/sd(abiotic$pH)
pH_anova_numeric <- lm(pH ~T+T:Stressor_numeric, data=abiotic)

pH_slope <- tidy(pH_anova_numeric,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")
pH_slope$std.error <- NULL
pH_slope$statistic <- NULL
pH_slope$p.value <- NULL

pH_slope$organisation="ecosystem"
pH_slope$variable="pH"

#TN

abiotic$TN <- (abiotic$TN-mean(abiotic$TN))/sd(abiotic$TN)
tn_anova_numeric <- lm(TN ~T+T:Stressor_numeric, data=abiotic)


tn_slope <- tidy(tn_anova_numeric,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")
tn_slope$std.error <- NULL
tn_slope$statistic <- NULL
tn_slope$p.value <- NULL

tn_slope$organisation="ecosystem"
tn_slope$variable="total nitrogen"

#tc 

abiotic$TC <- (abiotic$TC-mean(abiotic$TC))/sd(abiotic$TC)
tc_anova_numeric <- lm(TC ~T+T:Stressor_numeric, data=abiotic)

tc_slope <- tidy(tc_anova_numeric,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")
tc_slope$std.error <- NULL
tc_slope$statistic <- NULL
tc_slope$p.value <- NULL

tc_slope$organisation="ecosystem"
tc_slope$variable="total carbon"


#NMDS1

mds_whole_res$NMDS1 <- (mds_whole_res$NMDS1-mean(mds_whole_res$NMDS1))/sd(mds_whole_res$NMDS1)
anova_NMDS1_numeric <- lm(NMDS1 ~T+T:Stressor_numeric, data=mds_whole_res)


nmds1_slope <- tidy(anova_NMDS1_numeric,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")
nmds1_slope$std.error <- NULL
nmds1_slope$statistic <- NULL
nmds1_slope$p.value <- NULL

nmds1_slope$organisation="community"
nmds1_slope$variable="NMDS1"

#NMDS2

mds_whole_res$NMDS2 <- (mds_whole_res$NMDS2-mean(mds_whole_res$NMDS2))/sd(mds_whole_res$NMDS2)
anova_NMDS2_numeric <- lm(NMDS2 ~T+T:Stressor_numeric, data=mds_whole_res)



nmds2_slope <- tidy(anova_NMDS2_numeric,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")
nmds2_slope$std.error <- NULL
nmds2_slope$statistic <- NULL
nmds2_slope$p.value <- NULL

nmds2_slope$organisation="community"
nmds2_slope$variable="NMDS2"


#NMDS1_subset
mds_whole_res_subset$NMDS1 <- (mds_whole_res_subset$NMDS1-mean(mds_whole_res_subset$NMDS1))/sd(mds_whole_res_subset$NMDS1)
anova_NMDS1_mixed_numeric <- lm(NMDS1 ~T+T:Stressor_numeric, data=mds_whole_res_subset)

nmds1_subset_slope <- tidy(anova_NMDS1_mixed_numeric,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")
nmds1_subset_slope$std.error <- NULL
nmds1_subset_slope$statistic <- NULL
nmds1_subset_slope$p.value <- NULL


nmds1_subset_slope$organisation="community"
nmds1_subset_slope$variable="NMDS1_subpopulation"

#NMDS2_subset

mds_whole_res_subset$NMDS2 <- (mds_whole_res_subset$NMDS2-mean(mds_whole_res_subset$NMDS2))/sd(mds_whole_res_subset$NMDS2)
anova_NMDS2_mixed_numeric <- lm(NMDS2 ~T+T:Stressor_numeric, data=mds_whole_res_subset)
autoplot(anova_NMDS2_mixed_numeric)




nmds2_subset_slope <- tidy(anova_NMDS2_mixed_numeric,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")
nmds2_subset_slope$std.error <- NULL
nmds2_subset_slope$statistic <- NULL
nmds2_subset_slope$p.value <- NULL


nmds2_subset_slope$organisation="community"
nmds2_subset_slope$variable="NMDS2_subpopulation"


#Shannon

div_raw$Shannon <- (div_raw$Shannon-mean(div_raw$Shannon))/sd(div_raw$Shannon)
anova_diversity_numeric <- lm(Shannon ~T+T:Stressor_numeric, data=div_raw)
autoplot(anova_diversity_numeric)



shannon_slope <- tidy(anova_diversity_numeric,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")
shannon_slope$std.error <- NULL
shannon_slope$statistic <- NULL
shannon_slope$p.value <- NULL

shannon_slope$organisation="community"
shannon_slope$variable="species_richness"


#Tychonema

df_cyano$Abundance <- (df_cyano$Abundance-mean(df_cyano$Abundance))/sd(df_cyano$Abundance)
anova_tychonea_numeric <- lm(Abundance ~T+T:Stressor_numeric, data=df_cyano)
autoplot(anova_tychonea_numeric)

tychonema_slope <- tidy(anova_tychonea_numeric,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")
tychonema_slope$std.error <- NULL
tychonema_slope$statistic <- NULL
tychonema_slope$p.value <- NULL

tychonema_slope$organisation="species"
tychonema_slope$variable="Tychonema"


#Sulfuri

df_sulfuri$Abundance <- (df_sulfuri$Abundance-mean(df_sulfuri$Abundance))/sd(df_sulfuri$Abundance)
anova_sulfuri_numeric <- lm(Abundance ~T+T:Stressor_numeric, data=df_sulfuri)
autoplot(anova_sulfuri_numeric)

sulfuri_slope <- tidy(anova_sulfuri_numeric,conf.int = TRUE) %>%
  filter(term!="(Intercept)", term!="T24", term !="T28")
sulfuri_slope$std.error <- NULL
sulfuri_slope$statistic <- NULL
sulfuri_slope$p.value <- NULL

sulfuri_slope$organisation="species"
sulfuri_slope$variable="Sulfuricurvum"


all <- rbind(tychonema_slope,sulfuri_slope,shannon_slope,nmds1_slope,nmds2_slope,nmds1_subset_slope, nmds2_subset_slope,oxy_slope,pH_slope,tn_slope,tc_slope)


all <- all %>%
  reorder_levels(variable,order=c("Tychonema","Sulfuricurvum","NMDS1","NMDS2","NMDS1_subpopulation","NMDS2_subpopulation","species_richness","oxygen","pH","total nitrogen","total carbon"))


plot_slope <-all %>%
ggplot(aes(x = organisation, y = estimate, colour = variable, shape=term)) +
  geom_point(size = 4,position = position_dodge(width = 0.8))  +
  geom_hline(aes(yintercept=0))+
geom_errorbar(aes(x=organisation,y=estimate,col=variable,ymin=conf.low, ymax=conf.high), width=0.1,alpha=0.4,position = position_dodge(width = 0.8))+
  ylab("Slope [number of stressors]") +
    theme_bw() +
  scale_x_discrete(limits = c("species","community","ecosystem")) +
  xlab("Level of organisation") +
  theme(axis.title.x = element_text(size = 15), axis.text.x = element_text(size=12) ,axis.title.y = element_text(size = 15),axis.text.y = element_text(size=12))
  
```



